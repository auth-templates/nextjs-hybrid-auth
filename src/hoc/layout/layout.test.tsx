import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Layout from './layout';

describe('Layout', () => {
    it('contains title and logo', () => {
        render(
            <Layout title={"Title"}/>
        );

        expect(screen.getByRole("heading", "Title")).toBeInTheDocument();
        expect(screen.getByRole("img")).toBeInTheDocument();
    })

    test('if children is rendered', () => {
        render(
            <Layout>
                <div data-testid="text">text</div>
            </Layout>
        );

        expect(screen.getByTestId("text")).toBeInTheDocument();
    })

    it('renders correctly', () => {
        const tree = renderer.create(
            <Layout>
                <div data-testid="text">text</div>
            </Layout>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    })
})