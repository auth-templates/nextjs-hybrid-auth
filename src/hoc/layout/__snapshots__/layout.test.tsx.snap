// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders correctly 1`] = `
<div
  className="wrapper"
>
  <div
    className="logo"
  >
    <img
      alt="Company Logo"
      className="image"
      data-nimg="1"
      decoding="async"
      height={250}
      onError={[Function]}
      onLoad={[Function]}
      src="assets/images/dandelion.svg"
      style={
        {
          "color": "transparent",
        }
      }
      width={100}
    />
  </div>
  <h1
    className="title"
  />
  <div
    className="wrapper2"
  >
    <div
      data-testid="text"
    >
      text
    </div>
  </div>
</div>
`;
